torch/csrc/jit/codegen/cuda/parser.cpp

```diff
@@ -462,15 +462,14 @@ class IrParser {
             auto dims_list = constant_as<c10::List<int64_t>>(node->input(1));
             TORCH_INTERNAL_ASSERT(
                 dims_list.has_value(), "requires static reduce axes");
-            auto keepdim = constant_as<bool>(node->input(2));
             std::vector<int> dims;
             for (const auto dim : dims_list->vec()) {
               dims.emplace_back(static_cast<int>(dim));
             }
+            auto keepdim = constant_as<bool>(node->input(2));
             TORCH_INTERNAL_ASSERT(
-                keepdim.has_value() && !keepdim.value(),
-                "Keep dim in reduction is not a const false");
-            auto out = sum(self->as<TensorView>(), dims);
+                keepdim.has_value(), "requires static keepdim");
```

Comment:
Please make your assert messages a bit more holistic like, "Reductions cannot be fused without a static keepdim value."